source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -651,17 +663,17 @@ public Task<int> ExecuteNonQueryAsync(string query, CommandParameterValues? args
 
         public int ExecuteNonQuery(PreparedCommand preparedCommand)
         {
-            using var mutex = configuration.SupportConcurrentExecution 
-                ? DeadlockAwareLock.Lock()
+            using var mutex = configuration.ConcurrencyMode is not ConcurrencyMode.NoLock 
```

Comment:
Our no-lock ConcurrencyHandlers already stub out the Lock function, so why do we need this check at all?
Can we just `using var mutex = TransactionConcurrencyHandler.Lock()` and be done with it?